streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -463,9 +486,11 @@ private <VR> KTable<K, VR> doTransformValues(final ValueTransformerWithKeySuppli
 
         final String name = namedInternal.orElseGenerateWithPrefix(builder, TRANSFORMVALUES_NAME);
 
-        final KTableProcessorSupplier<K, V, K, VR> processorSupplier = new KTableTransformValues<>(
+        final FixedKeyProcessorSupplier<? super K, ? super V, ? extends VR> fixedKeyProcessorSupplier =
+            createValueTransformerWithKeySupplierToFixedProcessorSupplierAdaptor(transformerSupplier);
```

Comment:
To keep the refactoring internal, we put an adopter around the user provided "transformer supplier".